nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -143,23 +316,223 @@ public void process(final InputStream rawIn) throws IOException {
+                                                    new Object[]{ffFilename, bucket, key,
+                                                            currentState.getFilePosition(), skipped});
+                                        }
+                                    } catch (Exception e) {
+                                        getLogger().info("Failure skipping to resume upload flowfile={} bucket={} " +
+                                                "key={} position={} reason={}",
+                                                new Object[]{ffFilename, bucket, key, currentState.getFilePosition(),
+                                                        e.getMessage()});
+                                        throw(new ProcessException(e));
+                                    }
+                                }
+                            }
+
+                            // upload parts
+                            //------------------------------------------------------------
+                            long thisPartSize;
+                            for (int part = currentState.getPartETags().size() + 1;
+                                 currentState.getFilePosition() < currentState.getContentLength(); part++) {
+                                if (!PutS3Object.this.isScheduled()) {
+                                    getLogger().info("Processor unscheduled, stopping upload flowfile={} part={} " +
+                                            "uploadId={}", new Object[]{ffFilename, part, currentState.getUploadId()});
+                                    session.rollback();
+                                    return;
+                                }
+                                thisPartSize = Math.min(currentState.getPartSize(),
+                                        (currentState.getContentLength() - currentState.getFilePosition()));
+                                UploadPartRequest uploadRequest = new UploadPartRequest()
+                                        .withBucketName(bucket)
+                                        .withKey(key)
+                                        .withUploadId(currentState.getUploadId())
+                                        .withInputStream(in)
+                                        .withPartNumber(part)
+                                        .withPartSize(thisPartSize);
+                                try {
+                                    UploadPartResult uploadPartResult = s3.uploadPart(uploadRequest);
+                                    currentState.addPartETag(uploadPartResult.getPartETag());
+                                    currentState.setFilePosition(currentState.getFilePosition() + thisPartSize);
+                                    try {
+                                        persistState(cacheKey, currentState);
+                                    } catch (Exception e) {

```

Comment:
I'd add a comment here on why swallowing exceptions here is okay, as it is not immediately intuitive.